src/edge/edgeManager.ts

```diff
@@ -24,9 +24,18 @@ import { AmlManager } from "../container/amlManager";
 import { StreamAnalyticsManager } from "../container/streamAnalyticsManager";
 import { Marketplace } from "../marketplace/marketplace";
 
+enum ASAUpdateStatus {
+    Idle,
+    CheckingUpdate,
+    Updating,
+}
+
 export class EdgeManager {
 
+    private asaUpdateStatus: ASAUpdateStatus;

```

Comment:
move to asa manager?

Reply:
This would be used in the function:
![image](https://user-images.githubusercontent.com/5545529/63321578-ba7d8180-c353-11e9-9db8-33172da202e2.png)

This function also update module twin, it is better to put it in edgeManager

Reply:
OK